tests/pin.go

```diff
@@ -583,19 +583,10 @@ func assertNotPinned(t *testing.T, ctx context.Context, api iface.CoreAPI, p pat
 	}
 }
 
-func accPins(pins <-chan iface.Pin, err error) ([]iface.Pin, error) {
-	if err != nil {
-		return nil, err
-	}
-
+func accPins(pins <-chan iface.Pin, err <-chan error) ([]iface.Pin, error) {
 	var result []iface.Pin
-
 	for pin := range pins {
```

Comment:
so, when an error happen, the result channel is not closed right ? Doesn't that means the caller would get stuck in this reading loop ?

Reply:
With the new design, both the result and the error channels are _always_ closed.

Reply:
Hm. I need to document that.

Reply:
I've updated the issue description and the interface documentation. It looks like docker isn't actually all that consistent.